lib/hoodoo/services/middleware/middleware.rb

```
@@ -211,10 +211,37 @@ def self.memcached_host
     # Are we running on the queue, else (implied) a local HTTP server?
     #
     def self.on_queue?
-      q = ENV[ 'AMQ_ENDPOINT' ]
+      q = ENV[ 'AMQ_URI' ]
       q.nil? == false && q.empty? == false
     end
 
+    # Return a service 'name' derived from the service's collection of
+    # declared resources. The name will be the same across any instances of
+    # the service that implement the same resources. This can be used for
+    # e.g. AMQP-based queue-named operations, that want to target the same
+    # resource collection regardless of instance.
+    #
+    # This method will not work unless the middleware has parsed the set
+    # of service interface declarations (during instance initialisation).
+    # If a least one middleware instance has already been created, it is
+    # safe to call.
+    #
+    def self.service_name
+      @@service_name
```

If this isn't defined, things break with obscure errors. Arguably should raise a descriptive exception basically informing the caller that they called too early.